lib/Common/Memory/RecyclerPointers.h

```diff
@@ -145,10 +145,14 @@ template <>
 struct _ArrayItemWriteBarrierPolicy<_write_barrier_policy>
     { typedef _write_barrier_policy Policy; };
 
+template <class T, int N> // in case calling with type WriteBarrierPtr<void> [N]
```

Comment:
it's not matching that way... see my change in AuxPtrs. if there's
T arr[];
foo(&arr)

it's not going through WriteBarrierPtr<T>*

we can remove the '&' for sure, but just in case if there's such caller